pubsub/pubsub_goredis.go

```diff
@@ -139,8 +154,18 @@ func (ps *GoRedisPubSub) Subscribe(ctx context.Context, topic string, callback f
 				if msg == nil {
 					continue
 				}
+				receiveCtx, span := otelutil.StartSpanMulti(receiveRootCtx, ps.Tracer, "GoRedisPubSub.Receive", map[string]interface{}{
+					"topic":              topic,
+					"message_queue_size": len(redisch),
+					"message":            msg.Payload,
+				})
 				ps.Metrics.Count("redis_pubsub_received", 1)
-				go sub.cb(msg.Payload)
+
+				go func(cbCtx context.Context, span trace.Span) {
+					defer span.End()
+
+					sub.cb(cbCtx, msg.Payload)
+				}(receiveCtx, span)
```

Comment:
Why did you pass the context and span as parameters, but not the payload? I was originally thinking you needed none of them, but now, after thinking it through, it's probably a good idea to pass the message payload as well.

Reply:
Good catch, I just changed it